### PR TITLE
Allow building with OpenJDK 11u and vendor jlink plugins

### PR DESCRIPTION
**Pull request by jerboaa**
With [JDK-8232080](https://bugs.openjdk.java.net/browse/JDK-8232080) in progress of being backported to OpenJDK 11u the graal CE build with native image support is bound to fail. Failure looks like:

```
[Creating JDK image in /disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage]
/disk/openjdk/upstream-sources/openjdk-11-dev/build/linux-x86_64-normal-server-release/images/graal-builder-jdk/bin/jlink --add-modules=com.oracle.graal.graal_enterprise,java.base,java.compiler,java.datatransfer,java.desktop,java.instrument,java.logging,java.management,java.management.rmi,java.naming,java.net.http,java.prefs,java.rmi,java.scripting,java.se,java.security.jgss,java.security.sasl,java.smartcardio,java.sql,java.sql.rowset,java.transaction.xa,java.xml,java.xml.crypto,jdk.accessibility,jdk.aot,jdk.attach,jdk.charsets,jdk.compiler,jdk.crypto.cryptoki,jdk.crypto.ec,jdk.dynalink,jdk.editpad,jdk.hotspot.agent,jdk.httpserver,jdk.internal.ed,jdk.internal.jvmstat,jdk.internal.le,jdk.internal.opt,jdk.internal.vm.ci,jdk.internal.vm.compiler,jdk.internal.vm.compiler.management,jdk.internal.vm.compiler.truffle.jfr,jdk.jartool,jdk.javadoc,jdk.jcmd,jdk.jconsole,jdk.jdeps,jdk.jdi,jdk.jdwp.agent,jdk.jfr,jdk.jlink,jdk.jshell,jdk.jsobject,jdk.jstatd,jdk.localedata,jdk.management,jdk.management.agent,jdk.management.jfr,jdk.naming.dns,jdk.naming.rmi,jdk.net,jdk.pack,jdk.rmic,jdk.scripting.nashorn,jdk.scripting.nashorn.shell,jdk.sctp,jdk.security.auth,jdk.security.jgss,jdk.unsupported,jdk.unsupported.desktop,jdk.xml.dom,jdk.zipfs,org.graalvm.locator,org.graalvm.sdk,org.graalvm.truffle --module-path=/disk/graal/upstream-sources/graal/compiler/mxbuild/dists/jdk11/jdk.internal.vm.compiler.truffle.jfr.jmod:/disk/graal/upstream-sources/graal/sdk/mxbuild/dists/jdk11/org.graalvm.sdk.jmod:/disk/graal/upstream-sources/graal/truffle/mxbuild/dists/jdk11/org.graalvm.truffle.jmod:/disk/graal/upstream-sources/graal/compiler/mxbuild/dists/jdk11/jdk.internal.vm.compiler.jmod:/disk/graal/upstream-sources/graal/truffle/mxbuild/dists/jdk11/org.graalvm.locator.jmod:/disk/graal/upstream-sources/graal/compiler/mxbuild/dists/jdk11/jdk.aot.jmod:/disk/graal/upstream-sources/graal/compiler/mxbuild/dists/jdk11/jdk.internal.vm.compiler.management.jmod:/disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage.build/com.oracle.graal.graal_enterprise.jmod:/disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage.build/java.base.jmod:/disk/openjdk/upstream-sources/openjdk-11-dev/build/linux-x86_64-normal-server-release/images/graal-builder-jdk/jmods --output=/disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage -J-XX:+UseSerialGC -J-Xms32M -J-Xmx512M -J-XX:TieredStopAtLevel=1 -J-Dlink.debug=true --dedup-legal-notices=error-if-not-same-content --keep-packaged-modules=/disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage/jmods '--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UnlockExperimentalVMOptions' '--vendor-version=GraalVM 20.1.0-dev'
[Creating /disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage/lib/src.zip]
[Copying static libraries]
[Creating CDS shared archive]
/disk/graal/upstream-sources/graal/sdk/mxbuild/linux-amd64/graalvm-jimage/bin/java -Xshare:dump -Xmx128M -Xms128M
Unrecognized VM option 'EnableJVMCIProduct'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

Currently the build implies JDK-8232080 present => `-XX:+EnableJVMCIProduct` being present too which isn't the case for jdk11u-dev (upcoming 11.0.8).

I propose to fix this by adding a check whether this JVM option is indeed present, and only use it if so. Thoughts?

### CODE REVIEW

File: sdk/mx.sdk/mx_sdk_vm.py [hunk 1]
```diff
@@ -649,19 +661,22 @@ def jlink_new_jdk(jdk, dst_jdk_dir, module_dists,
...
         if jdk_has_new_jlink_options(jdk):
-            if jdk_omits_warning_for_jlink_set_ThreadPriorityPolicy(jdk):
-                thread_priority_policy_option = ' -XX:ThreadPriorityPolicy=1'
-            else:
-                mx.logv('[Creating JDK without -XX:ThreadPriorityPolicy=1]')
-                thread_priority_policy_option = ''
-
-            if any((m.name == 'jdk.internal.vm.compiler' for m in modules)):
-                jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
+            if jdk_supports_enablejvmciproduct(jdk):
```
Comment (dougxc) [inline sdk/mx.sdk/mx_sdk_vm.py:664]: I don't think `-XX:ThreadPriorityPolicy` is related to -XX:+EnableJVMCIProduct is it?

Comment (jerboaa) [reply sdk/mx.sdk/mx_sdk_vm.py:664]: No, but isn't `-XX:ThreadPriorityPolicy=1` intended for libgraal usage? If so, isn't `EnableJVMCIProduct` [enabling libgraal](https://bugs.openjdk.java.net/browse/JDK-8232118). That was my thinking anyway. Happy to change that, though.

Comment (dougxc) [reply sdk/mx.sdk/mx_sdk_vm.py:664]: No, setting `ThreadPriorityPolicy` is independent of libgraal.

Comment (jerboaa) [reply sdk/mx.sdk/mx_sdk_vm.py]: OK, thanks! Should be fixed now.

File: sdk/mx.sdk/mx_sdk_vm.py [hunk 2]
```diff
@@ -655,13 +667,16 @@ def jlink_new_jdk(jdk, dst_jdk_dir, module_dists,
...
+                    # Don't default to using JVMCI as JIT unless Graal is being updated in the image.
+                    # This avoids unexpected issues with using the out-of-date Graal compiler in
+                    # the JDK itself.
+                    jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UseJVMCICompiler -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
             else:
-                # Don't default to using JVMCI as JIT unless Graal is being updated in the image.
-                # This avoids unexpected issues with using the out-of-date Graal compiler in
-                # the JDK itself.
-                jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UseJVMCICompiler -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
+                mx.logv('[Creating JDK without -XX:+EnableJVMCIProduct]')
```
Comment (dougxc) [inline sdk/mx.sdk/mx_sdk_vm.py:679]: You need the following to make ThreadPriorityPolicy still be applied when EnableJVMCIProduct is not supported :
```
if thread_priority_policy_option:
    jlink.append('--add-options=' + thread_priority_policy_option.strip())
```

Comment (jerboaa) [reply sdk/mx.sdk/mx_sdk_vm.py:679]: Right, sorry about that. Should be fixed now. More thoughts?

Comment (dougxc) [reply sdk/mx.sdk/mx_sdk_vm.py:679]: looks good

Comment (jerboaa) [reply sdk/mx.sdk/mx_sdk_vm.py:679]: Thanks very much for the review, Doug!

**Pull request merged**